src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -176,7 +188,7 @@ private function parseDefinition($id, $service, $file)
         static::checkDefinition($id, $service, $file);
 
         if (isset($service['alias'])) {
-            $public = !array_key_exists('public', $service) || (bool) $service['public'];
+            $public = array_key_exists('public', $service) ? (bool) $service['public'] : !empty($defaults['public']);
```

Comment:
Any reason to define this variable twice? One doing `array_key_exists` other `isset`?